### PR TITLE
Extend the user, root and group creation tests

### PR DESCRIPTION
**Pull request by M4rtinK**
This should better cover what we provide support and expect in regards to user/root/groups creation.

### CODE REVIEW

File: rootpw-lock.ks.in [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#test name: rootpw-lock
+#
+# Test that roow account locking works correctly and
```
Comment (jkonecny12) [inline rootpw-lock.ks.in]: Typo `roow` -> `root`

File: rootpw-lock-no-password.ks.in [hunk 2]
```diff
@@ -0,0 +1,63 @@
+#test name: rootpw-lock-no-password
+#
+# Test that roow account locking works correctly and
```
Comment (jkonecny12) [inline rootpw-lock-no-password.ks.in]: The same typo.

File: fragments/shared/users/python3_check_root_locked.ks [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# check root account is locked
+%post --interpreter=/usr/bin/python3
+
+import sys
+import crypt
```
Comment (rvykydal) [inline fragments/shared/users/python3_check_root_locked.ks]: do we need to import crypt here?

File: rootpw-lock.ks.in (hunk 1 above)
Comment (M4rtinK) [reply rootpw-lock.ks.in]: Fixed.

File: rootpw-lock-no-password.ks.in (hunk 2 above)
Comment (M4rtinK) [reply rootpw-lock-no-password.ks.in]: Fixed as well. :)

File: fragments/shared/users/python3_check_root_locked.ks (hunk 3 above)
Comment (M4rtinK) [reply fragments/shared/users/python3_check_root_locked.ks]: Actually, good point, it got carried over from the original variant that also validated the password is correct, which unfortunately can't really be deduplicated via fragments due to needing to know the password.

Might be something to think about for the future - basically some argument/input passing for fragments, which would make more cases de-duplicable when the same check code needs to work on different kinds of input. 

Two ways com to my mind:
- some standardized files being dropped in place by previous snippet
- support for substitution inside of fragments in some way (things might become meta if we are not careful with this ;-) )

**Comment by M4rtinK** [comment]
@jkonecny12 
The new fragments are in use:

`$ git show | grep -H ksappend
:+%ksappend users/python3_check_root_locked.ks
:+%ksappend validation/success_if_result_empty_standalone.ks
:+%ksappend users/python3_check_root_locked.ks
:+%ksappend validation/success_if_result_empty_standalone.ks
:+%ksappend validation/success_if_result_empty_standalone.ks
:+%ksappend users/python3_check_root_locked.ks
:+%ksappend validation/success_if_result_empty_standalone.ks
:+%ksappend users/python3_check_root_locked.ks
:+%ksappend validation/success_if_result_empty_standalone.ks
`
Or did you mean something else ?

**Pull request merged**